contracts/peripheral/YieldDirector.sol

```diff
@@ -0,0 +1,390 @@
+        }
+
+        return total;
+    }
+
+    /************************
+    * Recipient Functions
+    ************************/
+
+    /**
+        @notice Get redeemable sOHM balance of a recipient address
+     */
+    function redeemableBalance(address recipient_) public override view returns (uint256) {
+        RecipientInfo storage recipient = recipientInfo[recipient_];
+        return recipient.carry
+            + _getAccumulatedValue(recipient.agnosticDebt, recipient.indexAtLastChange);
+    }
+
+    /**
+        @notice Redeem recipient's full donated amount of sOHM at current index
+        @dev Note that a recipient redeeming their vault shares effectively pays back all
+             sOHM debt to donors at the time of redeem. Any future incurred debt will
+             be accounted for with a subsequent redeem or a withdrawal by the specific donor.
+     */
+    function redeem() external override {
+        require(!redeemDisabled, "Redeems currently disabled");
+
+        uint256 redeemable = redeemableBalance(msg.sender);
+        require(redeemable > 0, "No redeemable balance");
+
+        RecipientInfo storage recipient = recipientInfo[msg.sender];
+        recipient.agnosticDebt = _toAgnostic(recipient.totalDebt);
+        recipient.carry = 0;
+        recipient.indexAtLastChange = IsOHM(sOHM).index();
+
+        IERC20(sOHM).safeTransfer(msg.sender, redeemable);
+
+        emit Redeemed(msg.sender, redeemable);
+    }
+
+    /************************
+    * Utility Functions
+    ************************/
+
+    /**
+        @notice Get accumulated sOHM since last time agnostic value changed.
+     */
+    function _getAccumulatedValue(uint256 gAmount_, uint256 indexAtLastChange_) internal view returns (uint256) {
+        return _fromAgnostic(gAmount_) - _fromAgnosticAtIndex(gAmount_, indexAtLastChange_);
+    }
+
+    /**
+        @notice Get array index of a particular recipient in a donor's donationInfo array.
+        @return Array index of recipient address. If recipient not present, returns max uint256 value.
+     */
+    function _getRecipientIndex(address donor_, address recipient_) internal view returns (uint256) {
+        DonationInfo[] storage info = donationInfo[donor_];
+
+        uint256 existingIndex = MAX_UINT256;
+        for (uint256 i = 0; i < info.length; i++) {
+            if(info[i].recipient == recipient_) {
+                existingIndex = i;
+                break;
+            }
+        }
+        return existingIndex;
+    }
+
+    /**
+        @notice Convert flat sOHM value to agnostic value at current index
+        @dev Agnostic value earns rebases. Agnostic value is amount / rebase_index.
+             1e9 is because sOHM has 9 decimals.
+     */
+    function _toAgnostic(uint256 amount_) internal view returns ( uint256 ) {
+        return amount_
+            * 1e9
+            / (IsOHM(sOHM).index());
```

Comment:
This is essentially converting the sOhm amount to gOhm amount right. gOhm has 18 decimals. Are you not concerned if the index becomes larger in future like let say 10000 by only using 9 decimals to store agnostic value we might lose some precision here?

Will probably be negligible amount but wanted to ask anyway.